### PR TITLE
cabal: Use Python 312

### PR DESCRIPTION
**Pull request by essandess**
* Fixes: https://trac.macports.org/ticket/69402

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification 
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: lang/cabal/Portfile [hunk 1]
```diff
@@ -4,7 +4,7 @@ PortSystem          1.0
 
 name                cabal
 version             3.10.2.0
-revision            2
+revision            3
```
Comment (reneeotten) [inline lang/cabal/Portfile:7]: no need to increase the `revision` - these are only affecting build dependencies.

Comment (essandess) [reply lang/cabal/Portfile]: Thanks, fixed.

File: python/py-sphinxnotes-strike/Portfile [hunk 2]
```diff
@@ -17,15 +17,20 @@ long_description    {*}${description}
...
 
 if {${name} ne ${subport}} {
-    depends_build-append \
+    if {${python.pep517}} {
+        depends_build-append \
+                    port:py${python.version}-setuptools_scm
+    } else {
+        depends_build-append \
                     port:py${python.version}-setuptools
+    }
```
Comment (reneeotten) [inline python/py-sphinxnotes-strike/Portfile:33]: `python.pep517` is always `yes` for Python 3.7+ subports, do this `if-else` block is unnecessary

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: python/py-sphinxnotes-strike/Portfile (hunk 2 above)
Comment (essandess) [reply python/py-sphinxnotes-strike/Portfile]: Thanks, fixed.

**Comment by essandess** [comment]
@reneeotten Are you able to see the CI issue? `port -vst desroot` works on my local system.

**Comment by essandess** [comment]
Dividing and conquering this CI issue. This now depends upon #22825 and #22826.

**Comment by reneeotten** [comment]
please rebase

**Comment by Schamschula** [comment]
It looks like both `py312-sphinx-jsonschema` and `py312-sphinxnotes-strike` don't exist. For that matter, neither has a `py311` or `py312` subport.

**Comment by essandess** [comment]
This appears to be an issue with the CI. `port -vst destroot cabal` works locally, and there's nothing in the logs that indicate an issue:

<details>
  <summary>find logs_21182654213 -type f -exec fgrep -i error {} ';'</summary>

    ### `find logs_21182654213 -type f -exec fgrep -i error {} ';'`
    ```sh
    $ find logs_21182654213 -type f -exec fgrep -i error {} ';'
    2024-02-27T22:05:34.8120620Z [36;1m    echo "::error::Failed to install dependencies for ${subport}"[0m
    2024-02-27T22:05:34.8139910Z [36;1m    echo "::error::Failed to install ${subport}"[0m
    2024-02-27T22:05:34.8144000Z [36;1m      -or -name CMakeError.log -or -name meson-log.txt); do[0m
    2024-02-27T22:05:34.8180790Z [36;1m    echo "::error::Tests failed for ${subport}"[0m
    2024-02-27T22:09:43.0846700Z ##[error]The operation was canceled.
    2024-02-27T22:05:33.0996780Z [36;1m    messagetype="error"[0m
    2024-02-27T22:04:45.8969410Z [36;1m    messagetype="error"[0m
    2024-02-27T22:04:49.5567330Z [36;1m    echo "::error::Failed to install dependencies for ${subport}"[0m
    2024-02-27T22:04:49.5582660Z [36;1m    echo "::error::Failed to install ${subport}"[0m
    2024-02-27T22:04:49.5588370Z [36;1m      -or -name CMakeError.log -or -name meson-log.txt); do[0m
    2024-02-27T22:04:49.5608260Z [36;1m    echo "::error::Tests failed for ${subport}"[0m
    2024-02-27T22:06:05.1554490Z ##[error]Failed to install dependencies for cabal-prebuilt
    2024-02-27T22:09:42.8442940Z ##[error]The operation was canceled.
    2024-02-27T22:05:33.0996790Z [36;1m    messagetype="error"[0m
    2024-02-27T22:05:34.8120620Z [36;1m    echo "::error::Failed to install dependencies for ${subport}"[0m
    2024-02-27T22:05:34.8139920Z [36;1m    echo "::error::Failed to install ${subport}"[0m
    2024-02-27T22:05:34.8144040Z [36;1m      -or -name CMakeError.log -or -name meson-log.txt); do[0m
    2024-02-27T22:05:34.8180790Z [36;1m    echo "::error::Tests failed for ${subport}"[0m
    2024-02-27T22:09:43.0852590Z ##[error]The operation was canceled.
    2024-02-27T22:03:58.1752090Z curl: (22) The requested URL returned error: 403
    2024-02-27T22:04:09.5490350Z [36;1m    messagetype="error"[0m
    2024-02-27T22:04:10.5055570Z [36;1m    echo "::error::Failed to install dependencies for ${subport}"[0m
    2024-02-27T22:04:10.5060780Z [36;1m    echo "::error::Failed to install ${subport}"[0m
    2024-02-27T22:04:10.5063310Z [36;1m      -or -name CMakeError.log -or -name meson-log.txt); do[0m
    2024-02-27T22:04:10.5070510Z [36;1m    echo "::error::Tests failed for ${subport}"[0m
    2024-02-27T22:04:44.2869950Z ##[error]Failed to install dependencies for cabal-prebuilt
    2024-02-27T22:09:29.3188570Z ##[error]Failed to install dependencies for cabal
    2024-02-27T22:09:29.3401490Z ##[error]Process completed with exit code 1.
    2024-02-27T22:04:30.9090020Z [36;1m    messagetype="error"[0m
    2024-02-27T22:04:33.2412270Z [36;1m    echo "::error::Failed to install dependencies for ${subport}"[0m
    2024-02-27T22:04:33.2422580Z [36;1m    echo "::error::Failed to install ${subport}"[0m
    2024-02-27T22:04:33.2427600Z [36;1m      -or -name CMakeError.log -or -name meson-log.txt); do[0m
    2024-02-27T22:04:33.2441850Z [36;1m    echo "::error::Tests failed for ${subport}"[0m
    2024-02-27T22:04:33.3320910Z df: getattrlist failed: Input/output error
    2024-02-27T22:04:33.3545230Z df: getattrlist failed: Input/output error
    2024-02-27T22:05:37.0732960Z df: getattrlist failed: Input/output error
    2024-02-27T22:05:39.1676170Z df: getattrlist failed: Input/output error
    2024-02-27T22:09:42.3725030Z ##[error]The operation was canceled.
    2024-02-27T22:04:49.5567290Z [36;1m    echo "::error::Failed to install dependencies for ${subport}"[0m
    2024-02-27T22:04:49.5582650Z [36;1m    echo "::error::Failed to install ${subport}"[0m
    2024-02-27T22:04:49.5588360Z [36;1m      -or -name CMakeError.log -or -name meson-log.txt); do[0m
    2024-02-27T22:04:49.5608220Z [36;1m    echo "::error::Tests failed for ${subport}"[0m
    2024-02-27T22:06:05.1554440Z ##[error]Failed to install dependencies for cabal-prebuilt
    2024-02-27T22:09:42.8442900Z ##[error]The operation was canceled.
    2024-02-27T22:04:45.8969400Z [36;1m    messagetype="error"[0m
    2024-02-27T22:04:10.5055560Z [36;1m    echo "::error::Failed to install dependencies for ${subport}"[0m
    2024-02-27T22:04:10.5060780Z [36;1m    echo "::error::Failed to install ${subport}"[0m
    2024-02-27T22:04:10.5063300Z [36;1m      -or -name CMakeError.log -or -name meson-log.txt); do[0m
    2024-02-27T22:04:10.5070500Z [36;1m    echo "::error::Tests failed for ${subport}"[0m
    2024-02-27T22:04:44.2869930Z ##[error]Failed to install dependencies for cabal-prebuilt
    2024-02-27T22:09:29.3188570Z ##[error]Failed to install dependencies for cabal
    2024-02-27T22:09:29.3401480Z ##[error]Process completed with exit code 1.
    2024-02-27T22:03:58.1752050Z curl: (22) The requested URL returned error: 403
    2024-02-27T22:04:09.5490340Z [36;1m    messagetype="error"[0m
    2024-02-27T22:04:33.2412260Z [36;1m    echo "::error::Failed to install dependencies for ${subport}"[0m
    2024-02-27T22:04:33.2422570Z [36;1m    echo "::error::Failed to install ${subport}"[0m
    2024-02-27T22:04:33.2427580Z [36;1m      -or -name CMakeError.log -or -name meson-log.txt); do[0m
    2024-02-27T22:04:33.2441840Z [36;1m    echo "::error::Tests failed for ${subport}"[0m
    2024-02-27T22:04:33.3320870Z df: getattrlist failed: Input/output error
    2024-02-27T22:04:33.3545210Z df: getattrlist failed: Input/output error
    2024-02-27T22:05:37.0732920Z df: getattrlist failed: Input/output error
    2024-02-27T22:05:39.1676140Z df: getattrlist failed: Input/output error
    2024-02-27T22:09:42.3725000Z ##[error]The operation was canceled.
    2024-02-27T22:04:30.9090010Z [36;1m    messagetype="error"[0m
    ```
</details>

**Pull request merged**